tls/s2n_x509_validator.c

```diff
@@ -503,9 +510,13 @@ static S2N_RESULT s2n_x509_validator_verify_cert_chain(struct s2n_x509_validator
 
     uint64_t current_sys_time = 0;
     RESULT_GUARD(s2n_config_wall_clock(conn->config, &current_sys_time));
+    if (sizeof(time_t) == 4) {
+        /* cast value to uint64_t to prevent overflow errors */
+        RESULT_ENSURE_LTE(current_sys_time, (uint64_t) MAX_32_TIMESTAMP_NANOS);
+    }
 
     /* this wants seconds not nanoseconds */
-    time_t current_time = (time_t) (current_sys_time / 1000000000);
+    time_t current_time = (time_t) (current_sys_time / ONE_SEC_IN_NANOS);
```

Comment:
Not totally sure that I'm following, but we have to compare the values as nanoseconds, otherwise the comparison would always be true.
```
RESULT_ENSURE_LTE(some_time_t, time_t_maximum_value)
```
Any `time_t` will necessarily be less than or equal to the max value of a `time_t`.

Reply:
Actually, maybe I'm wrong because I'm thinking of them both as `time_t` values, but in reality it's a macro that does who knows what with the types?

I'd prefer to keep the comparison as nanoseconds, because then I don't have to be clever and reason about typing/overflows in C macros.